gio/src/subclass/action.rs

```diff
@@ -0,0 +1,427 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use std::sync::OnceLock;
+
+use glib::{
+    ffi::{gboolean, GVariant, GVariantType},
+    prelude::*,
+    subclass::prelude::*,
+    translate::*,
+    GString, Variant, VariantType,
+};
+
+use crate::{ffi, Action};
+
+pub trait ActionImpl: ObjectImpl + ObjectSubclass<Type: IsA<Action>> {
+    fn name(&self) -> GString {
+        self.parent_name()
+    }
+
+    fn parameter_type(&self) -> Option<VariantType> {
+        self.parent_parameter_type()
+    }
+
+    fn state_type(&self) -> Option<VariantType> {
+        self.parent_state_type()
+    }
+
+    fn state_hint(&self) -> Option<Variant> {
+        self.parent_state_hint()
+    }
+
+    fn is_enabled(&self) -> bool {
+        self.parent_enabled()
+    }
+
+    fn state(&self) -> Option<Variant> {
+        self.parent_state()
+    }
+
+    fn change_state(&self, value: Variant) {
+        self.parent_change_state(value);
+    }
+
+    fn activate(&self, parameter: Option<Variant>) {
+        self.parent_activate(parameter);
+    }
+
+    fn delegate_get_property(
```

Comment:
These two delegates should probably go into the `Ext` trait. They're not supposed to be overridden, are they?

Reply:
Moved